crypto/provider_core.c

```diff
@@ -512,14 +512,21 @@ static int provider_free_intern(OSSL_PROVIDER *prov, int deactivate)
 }
 #endif
 
+OSSL_PROVIDER *ossl_provider_new(OSSL_LIB_CTX *libctx, const char *name,
+                                 OSSL_provider_init_fn *init_function,
+                                 int noconfig)
+{
+    return ossl_provider_new_ex(libctx, name, init_function, NULL, noconfig);
+}
+
 /*
  * We assume that the requested provider does not already exist in the store.
  * The caller should check. If it does exist then adding it to the store later
  * will fail.
  */
-OSSL_PROVIDER *ossl_provider_new(OSSL_LIB_CTX *libctx, const char *name,
-                                 OSSL_provider_init_fn *init_function,
-                                 int noconfig)
+OSSL_PROVIDER *ossl_provider_new_ex(OSSL_LIB_CTX *libctx, const char *name,
```

Comment:
Not sure why we are creating an "ex" version of an internal function. Just keep this as `ossl_provider_new` and convert all internal callers to the new param list.

Reply:
Done, thanks